test/functionalTest/cases/3727_avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself/avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself.test

```diff
@@ -0,0 +1,489 @@
+# Copyright 2023 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+avoid_triggering_notification_if_metadata_changes_but_not_the_attribute_value_itself
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. POST /v2/entities, to create entity E1 with attr A1 and metadata M1
+# 02. POST /v2/subscriptions, to create subscription for E1 with notifyOnMetadataChange=false
+# 03. PATCH /v2/entities/E1/attrs, to update the value of metadata
+# 04. Dump the accumulator, see no notification when metadata changes
+# 05. PATCH /v2/entities/E1/attrs, to update the value of attribute
+# 06. Dump the accumulator, see the notification when value of attribute changes
+# 07. POST /v2/entities, to create entity E2 with attr A1 and metadata M1
+# 08. POST /v2/subscriptions, to create subscription for E2 with notifyOnMetadataChange=true
+# 09. PATCH /v2/entities/E2/attrs, to update the value of metadata
+# 10. Dump the accumulator, see the notification when metadata changes
+# 11. PATCH /v2/entities/E2/attrs, to update the value of attribute
+# 12. Dump the accumulator, see the notification when value of attribute changes
+# 13. POST /v2/subscriptions, to create subscription for E3 with notifyOnMetadataChange=foo and see error
```

Comment:
I think these steps are fine, but I'd suggest to split into three different .test files, for the sake of maintainability. I mean:

* one .test file for notifyOnMetadataChange=true case
* one .test file for notifyOnMetadataChange=false case
* one .test file for notifyOnMetadataChange=foo case (error case)